app/lib/user_date_validator.rb

```diff
@@ -13,7 +13,12 @@ def child_date_of_birth_in_future
 
   def check_study_code
     codes = StudyCode.pluck(:title)
-    codes.include?(study_id)? true : errors.add(:study_id, "Study ID not present in our system. Please try with different one.") && false
+    if codes.all? { |code| Regexp.new(code).match(study_id) }
```

Comment:
The validation on `StudyCode` is such that it should contain exactly one record. So, arguably, the `if` statement's condition should be something like `Regexp.new(codes[0]).match(study_id)` (with an exception raised when `codes.length != 1`.